### PR TITLE
Added Mixpanel Destination

### PR DESCRIPTION
**Pull request by migs647**

### CODE REVIEW

File: Examples/destination_plugins/MixpanelDestination.swift [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            }
+            
+            mixpanel?.setGroup(groupKey: key, groupID: nsGroupID)
+            analytics?.log(message: "Mixpanel setGroup \(key) groupID \(groupID)")
+        }
+        
+        return event
+    }
+    
+    func alias(event: AliasEvent) -> AliasEvent? {
```
Comment (bsneed) [inline Examples/destination_plugins/MixpanelDestination.swift:173]: we've explicitly disabled alias in analytics-swift, fyi.  not sure how critical this bit is to overall operation.

File: Examples/destination_plugins/MixpanelDestination.swift [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        analytics?.log(message: "Mixpanel reset")
+    }
+    
+    func flush() {
+        mixpanel?.flush()
+        analytics?.log(message: "Mixpanel Flush")
+    }
+    
+    // MARK: - Private Methods
+    private func mixpanelTrack(_ eventName: String, properties: [String: Any]? = nil) {
```
Comment (bsneed) [inline Examples/destination_plugins/MixpanelDestination.swift:202]: might be good to move non-protocol conformance methods out into an extension.

**Review by bsneed — APPROVED** [review]

File: Examples/destination_plugins/MixpanelDestination.swift (hunk 2 above)
Comment (migs647) [reply Examples/destination_plugins/MixpanelDestination.swift:202]: Done

File: Examples/destination_plugins/MixpanelDestination.swift (hunk 1 above)
Comment (migs647) [reply Examples/destination_plugins/MixpanelDestination.swift:173]: Left for now... lets re-evaluate after discussions.

**Pull request merged**